### DISCUSSION TITLE
Foundry install

### DISCUSSION THREAD

**Discussion by EchoSpr** [Cyfrin/foundry-full-course-cu#30] [Q&A]
I installed foundry and followed the prompts. I checked to see that foundry was installed by checking the version and it was. However, after deleting the terminal and running ```foundryup ```again, it was as if foundry was not installed. How can I add foundry and make sure that it stays on my machine?

**Comment by cromewar** [comment]
When you install foundry you must add that to for path or environment variables, for example for Unix based system Mac & Linux, you should add something like this:

`export PATH="$PATH:/home/cromewar/.foundry/bin"`

to your `.zhrc` or `bashrc`.

If you are using windows you must add the path on which you installed foundry to for PATH tough windows advanced configuration.

please let us know is this is helpful for you.

**Reply by EchoSpr** [reply]
This turned into a bigger headache than I expected. Now every time that I try to install Foundry I am getting this error 
`bash: sed: command not found`  
I am also having trouble understanding how to configure my environment variables. Can someone do a screen share with me or help me better understand how to save these variables locally?

**Reply by EchoSpr** [reply]
This is how I added the path in my local environment, but for some reason, it is not being found.
`export PATH="$PATH:/Users/name/.foundry/bin"`

**Reply by cromewar** [reply]
From me seems a problem of a incorrect installation, try this:
1. use ->  which foundry.
2. Uninstall/Remove the folders containing foundry.
3. Install it again with: `curl -L https://foundry.paradigm.xyz | bash`
4. if this does not do it automatically add it manually to your `.bashrc` like this:

```bash
export PATH="$PATH:/Users/cromewar/.foundry/bin"

export PATH="$PATH:/Users/cromewar/.foundry/bin"
. "$HOME/.cargo/env"

```

replace cromewar with your username.

**Reply by Nkemjieme** [reply]
okay with a little of my idea i think u first try to check if u are on a visual environment and if so u can probably install it there but if not try using your cmd not from vscode to install it and add path to Ur environment variable if you're using windows but if it is mac u follow the process @cromewar  gave you in this discussion so try this two method out then u get to add it globally in your machine.

**Comment by EchoSpr** [comment]
I believe I got it configured now. But after running `foundryup` I am getting this error:

`curl: (23) Failed writing body
foundryup: command failed: curl -sSf https://api.github.com/repos/foundry-rs/foundry/git/refs/tags/nightly
foundryup: installing foundry (version nightly, tag nightly-)`

**Comment by PatrickAlphaC** [comment]
Hi @nickatnight23 

Could you run this command and then copy-paste to us what you get for an output?

```
curl -L https://foundry.paradigm.xyz | bash
```

run this, and then don't do anything until you get an output.

**Reply by EchoSpr** [reply]
This is what I got as an output. 

`bash: sed: command not found`

` % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1942  100  1942    0     0   5433      0 --:--:-- --:--:-- --:--:--  5433
Installing foundryup...
######################################################################## 100.0%`

`Detected your preferred shell is bash and added foundryup to PATH. Run 'source /Users/name/.bashrc' or start a new terminal session to use foundryup.
Then, simply run 'foundryup' to install Foundry.`

**Reply by EchoSpr** [reply]
Then after I do the next steps. I get the error that I posted above.

**Reply by PatrickAlphaC** [reply]
Great! Thank you!

Can you do the following steps for me?

1. Import the `source` line into your `.bash_profile` by running this command:
```
source /Users/name/.bashrc >> ~/.bash_profile
```
2. Close terminal (trash can icon!)
3. Re-open terminal, and run `foundryup`. If this fails send me your output
5. Close terminal again (trash can icon)
6. Then run`forge --help`

**Reply by EchoSpr** [reply]
This is the output: 

`/Users/name/.foundry/bin/foundryup: line 89: grep: command not found
/Users/name/.foundry/bin/foundryup: line 89: tr: command not found
/Users/name/.foundry/bin/foundryup: line 89: tr: command not found
/Users/name/.foundry/bin/foundryup: line 89: cut: command not found`

`curl: (23) Failed writing body
foundryup: command failed: curl -sSf https://api.github.com/repos/foundry-rs/foundry/git/refs/tags/nightly`

`foundryup: installing foundry (version nightly, tag nightly-)
/Users/name/.foundry/bin/foundryup: line 101: uname: command not found
foundryup: unsupported platform`

**Reply by cromewar** [reply]
@nickatnight23 this usually means that /usr/bin is not on the path or you have some missing dependencies which gets installed with XCode so try this:

1. If /usr/bin is not in the PATH, you can add it temporarily by running export PATH=$PATH:/usr/bin. Try running the foundry installation again.
2. If the above steps work, you will need to make sure the PATH change is permanent, which typically involves adding the export PATH=$PATH:/usr/bin line to a startup script such as ~/.bashrc, ~/.bash_profile, or ~/.zshrc (depending on your shell).
3. If the issue persists, it's possible that these utilities somehow got removed from your system. You can check if they're present by typing /usr/bin/grep --version (and similarly for tr, cut, and uname). If they're not present, you may need to reinstall them. The process for doing so will depend on your system. For example, on macOS, you might need to reinstall the Xcode command line tools with xcode-select --install.

**Reply by EchoSpr** [reply]
Thank you everyone!  It seems like it is working now. I am still having this error in the command line `bash: sed: command not found` However, I am also getting this output after entering `forge --version` `forge 0.2.0 (08a629a 2023-06-03T00:14:26.563741000Z)` which means it is installed and working!

**Reply by PatrickAlphaC** [reply]
Awesome!! so it's at least downloaded! Are you able to run the commands I talk about in the rest of the video?

**Reply by EchoSpr** [reply]
Yes, I was! Thank you so much!
